integration-tests/google-storage/src/test/java/org/apache/camel/quarkus/component/google/storage/it/GoogleStorageTest.java

```diff
@@ -0,0 +1,183 @@
+                        .get("/google-storage/stopRoute")
+                        .then()
+                        .statusCode(200);
+            }
+        }
+    }
+
+    @Test
+    public void testProducer() {
+        //delete existing buckets t - only on real account - Deleting buckets is not (yet) supported by fsouza/fake-gcs-server.
+        if (!MockBackendUtils.startMockBackend()) {
+            String buckets = executeOperation(GoogleCloudStorageOperations.listBuckets, Collections.emptyMap(),
+                    null);
+            List<String> bucketsToDelete = Arrays.stream(buckets.split(","))
+                    .filter(b -> b.equals(TEST_BUCKET1) || b.equals(TEST_BUCKET2))
+                    .collect(Collectors.toList());
+            if (!bucketsToDelete.isEmpty()) {
+                bucketsToDelete.forEach(
+                        b -> executeOperation(b, GoogleCloudStorageOperations.deleteBucket, Collections.emptyMap(),
+                                is(Boolean.toString(true))));
+            }
+        }
+
+        //create object in testBucket
+        putObject("Sheldon", TEST_BUCKET1, FILE_NAME_007);
+
+        putObject("Irma", TEST_BUCKET2, FILE_NAME_006);
+
+        //copy object to test_bucket2
+        executeOperation(GoogleCloudStorageOperations.copyObject,
+                CollectionHelper.mapOf(GoogleCloudStorageConstants.OBJECT_NAME, FILE_NAME_007,
+                        GoogleCloudStorageConstants.DESTINATION_BUCKET_NAME, TEST_BUCKET2,
+                        GoogleCloudStorageConstants.DESTINATION_OBJECT_NAME, FILE_NAME_007 + "_copy"),
+                is("Sheldon"));
+
+        //GetObject
+        executeOperation(TEST_BUCKET2, GoogleCloudStorageOperations.getObject,
+                Collections.singletonMap(GoogleCloudStorageConstants.OBJECT_NAME, FILE_NAME_007 + "_copy"),
+                is("Sheldon"));
+
+        //list buckets
+        executeOperation(GoogleCloudStorageOperations.listBuckets, Collections.emptyMap(),
+                both(containsString(TEST_BUCKET1)).and(containsString(TEST_BUCKET2)));
+
+        //deleteObject
+        executeOperation(TEST_BUCKET2, GoogleCloudStorageOperations.deleteObject,
+                CollectionHelper.mapOf(GoogleCloudStorageConstants.OBJECT_NAME, FILE_NAME_006),
+                is(Boolean.toString(true)));
+
+        //ListObjects
+        executeOperation(TEST_BUCKET2, GoogleCloudStorageOperations.listObjects, Collections.emptyMap(),
+                containsString(FILE_NAME_007 + "_copy"));
+
+        //CreateDownloadLink - requires authentication
+        if (!GoogleStorageHelper.usingMockBackend()) {
+            executeOperation(TEST_BUCKET2, GoogleCloudStorageOperations.createDownloadLink,
+                    Collections.singletonMap(GoogleCloudStorageConstants.OBJECT_NAME, FILE_NAME_007 + "_copy"),
+                    startsWith("http"));
+        }
+    }
+
+    private void putObject(String sheldon, String testBucket1, String fileName007) {
```

Comment:
Maybe the param names could be more generic here. testBucket ? filename ?

Reply:
Fixed, probably generated method and I didn't notice wrong param names.